core/domain/exp_domain.py

```diff
@@ -3060,90 +3172,1449 @@ def _convert_states_v51_dict_to_v52_dict(
-                    content_id_list.extend(
-                        customisation_args[ca_name].get_content_ids()
-                    )
+                collapsible_content_list = json.loads(
+                    collapsible_content_json)
+                if len(collapsible_content_list) == 0:
+                    tag.decompose()
+
+                collapsible_content_list = cls.fix_rte_tags(
+                    collapsible_content_list,
+                    is_tags_nested_inside_tabs_or_collapsible=True
+                )
+                collapsible_content_json = json.dumps(collapsible_content_list)
+                tag['content-with-value'] = utils.escape_html(
+                    collapsible_content_json)
+            else:
+                tag.decompose()
+
+            if tag.has_attr('heading-with-value'):
+                collapsible_heading_json = (
+                    utils.unescape_html(tag['heading-with-value']))
+                collapsible_heading_list = json.loads(
+                    collapsible_heading_json)
+                if len(collapsible_heading_list) == 0:
+                    tag.decompose()
+            else:
+                tag.decompose()
 
-            translations_mapping = (
-                state_dict['written_translations']['translations_mapping'])
-            new_translations_mapping = {}
-            for content_id, translation_item in translations_mapping.items():
-                if content_id in content_id_list:
-                    new_translations_mapping[content_id] = translation_item
-            state_dict['written_translations']['translations_mapping'] = (
-                new_translations_mapping)
-
-            voiceovers_mapping = (
-                state_dict['recorded_voiceovers']['voiceovers_mapping'])
-            new_voiceovers_mapping = {}
-            for content_id, voiceover_item in voiceovers_mapping.items():
-                if content_id in content_id_list:
-                    new_voiceovers_mapping[content_id] = voiceover_item
-            state_dict['recorded_voiceovers']['voiceovers_mapping'] = (
-                new_voiceovers_mapping)
+        return str(soup).replace('<br/>', '<br>')
 
+    @classmethod
+    def _update_state_rte(
+        cls,
+        states_dict: Dict[str, state_domain.StateDict]
+    ) -> Dict[str, state_domain.StateDict]:
+        """Update the state RTE content and translations
+
+        Args:
+            states_dict: dict. A dict where each key-value pair represents,
+                respectively, a state name and a dict used to initialize a
+                State domain object.
+
+        Returns:
+            dict. The converted states_dict.
+        """
+        for state in states_dict.values():
+            # Fix tags for state content.
+            html: str = state['content']['html']
+            html = cls.fix_rte_tags(html)
+            html = cls.fix_tabs_and_collapsible_tags(html)
+            state['content']['html'] = html
```

Comment:
You can isolate this into a helper function and then use it here an below.

Reply:
Done